deep_toggl/config.py

```diff
@@ -66,41 +64,41 @@ def get_all_tags(self):
         return result
 
 
-def setup():
-    config = configparser.ConfigParser()
-    token = get_api_token()
-    tg = TogglWrapper(token)
-    projects = tg.get_all_projects()
-    tags = tg.get_all_tags()
-
-    projects_names = [p.name for p in projects]
-    tags_names = [x.name for x in tags]
-    tags_names.append(ANY_TAG)
-
-    questions = [
-        inquirer.Checkbox(
-            "projects", message="Which projects to track?", choices=projects_names
-        ),
-    ]
-    answers = inquirer.prompt(questions, theme=GreenPassion())
-    projects_chosen = answers["projects"]
-
-    for project_chosen in projects_chosen:
+def get_chosen_projects(toggl_wrapper: TogglWrapper) -> List[str]:
+    projects_names = [x.name for x in toggl_wrapper.get_all_projects()]
+    while True:
         questions = [
-            inquirer.Checkbox(
-                "tags",
-                message=f"What tags to track for {project_chosen}?",
-                choices=tags_names,
-            )
+            inquirer.Checkbox("projects", message="Which projects to track?", choices=projects_names,),
         ]
         answers = inquirer.prompt(questions, theme=GreenPassion())
+        projects_chosen = answers["projects"]
+        if len(projects_chosen) != 0:
+            return projects_chosen
+        print("You have to pick at least one project.")
+
+
+def get_chosen_tags(toggl_wrapper: TogglWrapper, chosen_project: str) -> List[str]:
+    tags_names = [ANY_TAG]
+    tags_names.extend(x.name for x in toggl_wrapper.get_all_tags())
+    questions = [inquirer.Checkbox("tags", message=f"What tags to track for {chosen_project}?", choices=tags_names,)]
+    answers = inquirer.prompt(questions, theme=GreenPassion())
+    return answers["tags"]
+
+
+def setup():
```

Comment:
Just a comment. I think the review process is easier if we update functions "in loco" instead of removing them and declaring them in another point of the file. What do you think? @albertodvp

Reply:
Absolutely. I didn't check it. I will be more careful